metrics.go

```diff
@@ -209,6 +182,29 @@ func (m *metrics) sendMetrics() {
 	m.sent <- payload
 }
 
+func (m *metrics) doPost(url *url.URL, payload interface{}) (*http.Response, error) {
+	var body bytes.Buffer
+	enc := json.NewEncoder(&body)
+	if err := enc.Encode(payload); err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest("POST", url.String(), &body)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Add("Unleash-Appname", m.options.appName)
+	req.Header.Add("Unleash-Instanceid", m.options.instanceId)
```

Comment:
I'll change this to be consistent with the other clients. Go will internally send the canonical form with the request, but I guess the server handles this.